src/Tests/ContainerGroup.fs

```diff
@@ -138,5 +139,51 @@ let tests = testList "Container Group" [
         Expect.isNotNull group.Volumes.[3].GitRepo "Git repo volume should not be null"
     }
 
-]
+    test "Container group with system assigned identity" {
+        let group =
+            containerGroup {
+                name "myapp"
+                add_instances [ nginx ]
+                identity SystemAssigned
+            } |> asAzureResource
+
+        Expect.isTrue group.Identity.Type.HasValue "Expecting an assigned identity."
+        Expect.equal group.Identity.Type.Value ResourceIdentityType.SystemAssigned "Expecting a system assigned identity"
+    }
+    
+    test "Container group with user assigned identity" {
+        let group =
+            containerGroup {
+                name "myapp"
+                add_instances [ nginx ]
+                identity (UserAssigned [ ManagedIdentity.UserAssignedIdentity("foo", Some "bar")])
+            } |> asAzureResource
+
+        Expect.hasLength group.Identity.UserAssignedIdentities 1 "No user assigned identity."
+    }
+    
+    test "Make container group with MSI" {
+        let msi = userAssignedIdentity {
+            name "aciUser"
+        }
+        let group =
+            containerGroup {
+                name "myapp-with-msi"
+                add_instances [ nginx ]
+                user_assigned_identity msi.Name
+            }
+        let template = arm {
+            location Location.EastUS
+            add_resource msi
```

Comment:
Again, from a consumption point of view - will people expect that they need to explicitly add this resource to the template? Or should it simply be added when injected into the `containerGroup`?